src/lib/utils/environment.ts

```diff
@@ -0,0 +1,20 @@
+import { hosts } from '@/lib/constants/environment'
+
+export const generateAbsoluteUrlFromBuildType = (
+  relativeUrl: string,
+  buildType: string
+) => {
+  const host = hosts[buildType]
+  if (!host) {
+    return relativeUrl
+  }
+  return relativeUrl.charAt(0) === '/'
+    ? `${host.origin}${relativeUrl}`
+    : `${host.origin}/${relativeUrl}`
+}
+
+export const generateAbsoluteUrlFromEnv = (relativeUrl: string) =>
+  generateAbsoluteUrlFromBuildType(
+    relativeUrl,
+    process.env.NEXT_PUBLIC_BUILD_TYPE
```

Comment:
i think we may want to use `APP_ENV` over `NEXT_PUBLIC_BUILD_TYPE`

Reply:
See [my other comment](https://github.com/department-of-veterans-affairs/next-build/pull/269#discussion_r1415880770). I'm not sure either of these is ultimately what we want. I think this is a bit deeper. The rhetorical question I would ask is, what is the value of `APP_ENV` on Tugboat?

Reply:
it's `tugboat`, so naively I thought something like:

```
  TUGBOAT: 'tugboat',

  [environments.TUGBOAT]: {
    hostname: process.env.TUGBOAT_DEFAULT_SERVICE_URL_HOST,
    origin: process.env.TUGBOAT_DEFAULT_SERVICE_URL,
  },
```

Reply:
This is clarifying. I need to chew on things still, but this helps paint the picture of your vision.